compiler/definitions/ir/resource.py

```diff
@@ -56,3 +57,49 @@ def __eq__(self, other):
 class EphemeralResource(Resource):
     def __init__(self):
         self.uri = None
+
+class RemoteFileResource(Resource):
+    def __init__(self):
+        raise NotImplementedError("RemoteFileResource is an interface")
+
+    def __str__(self):
+        raise NotImplementedError("RemoteFileResource is an interface")
+
+    def is_available_on(self, host):
+        raise NotImplementedError("RemoteFileResource is an interface")
+
+    def _normalize_addr(self, addr):
+        """
+        Helper, removes port number if supplied in address and normalizes host address
+        Example1: datanode1 -> 172.18.0.3
+        Example2: 172.18.0.3:5555 -> 172.18.0.3
+        """
+        host = addr.rsplit(":", 1)[0]
+        normalized_host = socket.gethostbyaddr(host)[2][0]
+        return normalized_host
+
+class HDFSFileResource(RemoteFileResource):
+    ## The uri is the path of the file.
+    def __init__(self, uri, resource_hosts):
+        """
+        Params:
+            uri: Usually the path to the file
```

Comment:
Is the path to the file the same in all machines? If not, let's add a TODO here to fix that later.

Reply:
The path is the same for hdfs; the top directory however could be different. That's why the str for HDFSResource returns "$HDFS_DATANODE_DIR/{self.uri}". I set the HDFS_DATANODE_DIR in the worker's environment.

Reply:
Great, a comment on that would be nice :)